### PR TITLE
#128: integrate SortTable to markets

### PR DESCRIPTION
**Pull request by lapinlapin**
#### What's this PR do?
Rewrite markest for table sort
Edit markup and some logic from redesign
#### Where should the reviewer start?
views/TheMarkets/TheMarkets.vue
#### How should this be manually tested?
http://localhost:8080/#/
#### What are the relevant tickets?
no tickets
#### Screenshots (if appropriate)
image

#### URL
Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/128-sortable-markets

### CODE REVIEW

File: src/components/SortableHeaderItem/SortableHeaderItem.vue [hunk 1]
```diff
@@ -61,6 +65,12 @@ export default {
   &--left-aligned {
     justify-content: flex-start;
   }
+  &--center-aligned {
+    text-align: center;
```
Comment (roma219) [inline src/components/SortableHeaderItem/SortableHeaderItem.vue]: why text-align? `justify-content: center` will do it

File: src/views/TheMarkets/MarketsTickersList.vue [hunk 2]
```diff
@@ -1,45 +1,40 @@
...
+      </template>
+    </SortableTable>
   </div>
 </template>
 
 <script>
 import { mapActions } from 'vuex'
 import MarketsTickersListItem from './MarketsTickersListItem'
+import SortableTable from '@/components/SortableTable'
 import SortableHeaderItem from '@/components/SortableHeaderItem'
```
Comment (roma219) [inline src/views/TheMarkets/MarketsTickersList.vue]: SortableHeaderItem is not used anymore, right?

File: src/views/TheMarkets/MarketsTickersList.vue [hunk 3]
```diff
@@ -66,22 +61,25 @@ export default {
   },
   data() {
     return {
-      sortField: 'ticker',
-      sortType: 'asc',
+      sortField: {
```
Comment (roma219) [inline src/views/TheMarkets/MarketsTickersList.vue:62]: Only used for initial sorting now, right?

Comment (roma219) [reply src/views/TheMarkets/MarketsTickersList.vue:62]: There is a `:default-sort` prop in `SortableTable`

File: src/views/TheMarkets/MarketsTickersList.vue [hunk 4]
```diff
@@ -90,116 +88,44 @@ export default {
...
 </script>
 
 <style lang="scss">
   .tickers-list {
-    margin-top: 0.5rem;
+    margin-top: 0.8rem;
+    margin-left: 0.4375rem;
     font-weight: config('fontWeights.semibold');
-
+    //fix for table view
```
Comment (roma219) [inline src/views/TheMarkets/MarketsTickersList.vue]: Please no styles for any child components inside a parent component styles

File: tailwind.js [hunk 5]
```diff
@@ -45,6 +45,7 @@ let colors = {
   'transparent': 'transparent',
 
   'black': '#22292f',
+  'grey-ticker': 'rgb(255, 255, 255, .5)',
```
Comment (roma219) [inline tailwind.js]: You can use `colors.inactive` instead

**Review by roma219 — CHANGES_REQUESTED** [review]
Well done, but a few keypoint that need to be changes:
1) Aligning inside HeaderItem can be done with `justify-content: center` - this will work if you remove you custom styles for HeaderItem from you TableComponent - they also make header title jump to the left when sorting (compare to how it is displayed in Account)
2) And now everything will be good except for the left column. Possible solution: left column should be aligned to the left, make a big padding for table, so there is a space on the left. Then position star in every Item component with position absolulte, with `-` left position.

File: src/views/TheMarkets/MarketsTickersList.vue (hunk 3 above)
Comment (lapinlapin) [reply src/views/TheMarkets/MarketsTickersList.vue:62]: yes

**Review by roma219 — APPROVED** [review]

**Pull request merged**